From: Ben Hutchings <ben@decadent.org.uk>
Date: Sat, 6 Dec 2014 22:47:25 +0000
Subject: [PATCH 8/9] net: sendmsg: Really fix NULL pointer dereference

The backport of upstream commit 40eea803c6b2 ("net: sendmsg: fix NULL
pointer dereference") in 2.6.32.64 incorrectly tested msg_namelen
twice rather than both msg_name and msg_namelen.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 net/compat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/compat.c b/net/compat.c
index 71ed839..a5848ac 100644
--- a/net/compat.c
+++ b/net/compat.c
@@ -83,7 +83,7 @@ int verify_compat_iovec(struct msghdr *kern_msg, struct iovec *kern_iov,
 {
 	int tot_len;
 
-	if (kern_msg->msg_namelen && kern_msg->msg_namelen) {
+	if (kern_msg->msg_name && kern_msg->msg_namelen) {
 		if (mode==VERIFY_READ) {
 			int err = move_addr_to_kernel(kern_msg->msg_name,
 						      kern_msg->msg_namelen,
